graph/manifest.go

```diff
@@ -75,12 +75,15 @@ func (s *TagStore) newManifest(localName, remoteName, tag string) ([]byte, error
 
 			tarSum, err := tarsum.NewTarSum(archive, true, tarsum.Version1)
 			if err != nil {
+				archive.Close()
```

Comment:
Reading more about defer. Found following.

http://blog.golang.org/defer-panic-and-recover

"A deferred function's arguments are evaluated when the defer statement is evaluated".

That probably means that I can just call "defer archive.Close()" and this can happen in loop and all defer instances should be able to figure out which archive they should be calling close() on when function finishes.

If that works, then only issue remaining will be that all these defer will be called when function returns and devices will remain mounted. Hopefully there is proper reference counting in graph driver backend which can keep count of nested Get() and Put() operations.